core/ir/instr_api.h

```diff
@@ -298,10 +298,14 @@ struct _instr_t {
 
     uint opcode;
 
-#    ifdef X86
-    /* PR 251479: offset into instr's raw bytes of rip-relative 4-byte displacement */
-    byte rip_rel_pos;
-#    endif
+    union {
+        /* PR 251479: offset into instr's raw bytes of rip-relative 4-byte displacement */
+        byte rip_rel_pos;
+        /* PR #6691: size of source data (i.e., read) an instruction operates on.
+         * Note that opnd_size_t is an alias for byte.
+         */
+        opnd_size_t operation_size;
```

Comment:
Explain when this field is valid as opposed to rip_rel_pos.

Reply:
Done.